An Institution wishes to find out their student’s ability in mathematics, reading
and writing skills. The Institution wants to do an exploratory study to check the
following information.
1. Find out how many males and females participated in the test.
2. What do you think about the students' parental level of education?
3. Who scores the most on average for math, reading and writing based on\

     ● Gender

     ● Test preparation course
4. What do you think about the scoring variation for math, reading and writing
based on

    ● Gender

    ● Test preparation course
5. The management needs your help to give bonus points to the top 25% of
students based on their math score, so how will you help the management
to achieve this.

In [1]:
import numpy as np

In [2]:
import pandas as pd

In [3]:
Student_Performance=pd.read_csv('/content/StudentsPerformance.csv')

In [4]:
Student_Performance

gender race/ethnicity parental level of education         lunch  \
0    female        group C            some high school  free/reduced   
1    female        group B                 high school  free/reduced   
2    female        group B            some high school  free/reduced   
3    female        group B                some college      standard   
4    female        group C                some college  free/reduced   
..      ...            ...                         ...           ...   
995    male        group E                some college      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
998    male        group E          associate's degree  free/reduced   
999    male        group E           bachelor's degree      standard   

    test preparation course  math score  reading score  writing score  
0                      none           0             17             10  
1                      none           8             24             23  
2                      none          18             32             28  
3                      none          11             38             32  
4                      none          22             39             33  
..                      ...         ...            ...            ...  
995               completed          99             87             81  
996               completed         100             96             86  
997               completed         100             97             99  
998               completed         100            100             93  
999               completed         100            100            100  

[1000 rows x 8 columns]

In [ ]:
import statistics as stat

1.Find out how many males and females participated in the test.

In [5]:
Student_Performance['gender'].value_counts()

female    518
male      482
Name: gender, dtype: int64

In [6]:
Student_Performance.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 8 columns):
 #   Column                       Non-Null Count  Dtype 
---  ------                       --------------  ----- 
 0   gender                       1000 non-null   object
 1   race/ethnicity               1000 non-null   object
 2   parental level of education  1000 non-null   object
 3   lunch                        1000 non-null   object
 4   test preparation course      1000 non-null   object
 5   math score                   1000 non-null   int64 
 6   reading score                1000 non-null   int64 
 7   writing score                1000 non-null   int64 
dtypes: int64(3), object(5)
memory usage: 62.6+ KB


In [7]:
Student_Performance.describe(include='all')

gender race/ethnicity parental level of education     lunch  \
count     1000           1000                        1000      1000   
unique       2              5                           6         2   
top     female        group C                some college  standard   
freq       518            319                         226       645   
mean       NaN            NaN                         NaN       NaN   
std        NaN            NaN                         NaN       NaN   
min        NaN            NaN                         NaN       NaN   
25%        NaN            NaN                         NaN       NaN   
50%        NaN            NaN                         NaN       NaN   
75%        NaN            NaN                         NaN       NaN   
max        NaN            NaN                         NaN       NaN   

       test preparation course   math score  reading score  writing score  
count                     1000  1000.000000    1000.000000    1000.000000  
unique                       2          NaN            NaN            NaN  
top                       none          NaN            NaN            NaN  
freq                       642          NaN            NaN            NaN  
mean                       NaN    66.001000      69.195000      68.119000  
std                        NaN    15.569567      14.706436      15.226381  
min                        NaN     0.000000      17.000000      10.000000  
25%                        NaN    57.000000      59.000000      58.000000  
50%                        NaN    66.000000      70.000000      69.000000  
75%                        NaN    77.000000      79.250000      79.000000  
max                        NaN   100.000000     100.000000     100.000000

2.What do you think about the students' parental level of education?

In [8]:
Student_Performance["parental level of education"].value_counts()

some college          226
associate's degree    222
high school           196
some high school      179
bachelor's degree     118
master's degree        59
Name: parental level of education, dtype: int64

3. Who scores the most on average for math, reading and writing based on

● Gender

● Test preparation course

In [9]:
#Gender
Student_Performance.groupby('gender')[['math score','reading score','writing score']].mean()

math score  reading score  writing score
gender                                          
female   63.376448      72.590734      72.467181
male     68.821577      65.545643      63.446058

Female scores the most on all these fields with an average of 63.376448 for math and 72.590734 for reading and for writing it is 72.467181

In [10]:
# Test prepartion course
Student_Performance.groupby('test preparation course')[['math score','reading score','writing score']].mean()

math score  reading score  writing score
test preparation course                                          
completed                  69.96648      74.175978      74.684358
none                       63.78972      66.417445      64.457944

The persons who have completed have the highest average score.

  69.96648	is the average math score for who have completed test prepartion

  74.175978 is the average reading score for who have completed test prepartion
  
  74.684358 is the average writing  score for who have completed test prepartion

4 .What do you think about the scoring variation for math, reading and writing based on

● Gender

● Test preparation course

In [11]:
# Gender
Student_Performance.groupby('gender')[['math score','reading score','writing score']].var()

math score  reading score  writing score
gender                                          
female  256.958593     207.677438     220.369327
male    211.889097     200.211010     202.413924

In [12]:
#Standard deviation
Student_Performance.groupby('gender')[['math score','reading score','writing score']].std()

math score  reading score  writing score
gender                                          
female   16.029928      14.411018      14.844842
male     14.556411      14.149594      14.227225

In [13]:
# min
Student_Performance.groupby('gender')[['math score','reading score','writing score']].min()

math score  reading score  writing score
gender                                          
female           0             17             10
male            27             23             15

In [14]:
#max
Student_Performance.groupby('gender')[['math score','reading score','writing score']].max()

math score  reading score  writing score
gender                                          
female         100            100            100
male           100            100            100

Range=max- min

 Female :
 Math score-100-0=0
 reading score-100-17=83
 Writing score-100-10=90

 Male ;
 Math score-100-27=73
 reading score-100-23=77
 Writing score-100-15=85


In [15]:
#Preparation course - variance
Student_Performance.groupby('test preparation course')[['math score','reading score','writing score']].var()

math score  reading score  writing score
test preparation course                                          
completed                210.884027     183.265864     175.202612
none                     246.668662     213.419851     226.251739

In [16]:
# standard deviation
Student_Performance.groupby('test preparation course')[['math score','reading score','writing score']].std()

math score  reading score  writing score
test preparation course                                          
completed                 14.521847      13.537572      13.236412
none                      15.705689      14.608896      15.041667

In [17]:
Student_Performance.groupby('test preparation course')[['math score','reading score','writing score']].min()

math score  reading score  writing score
test preparation course                                          
completed                        23             37             36
none                              0             17             10

In [18]:
Student_Performance.groupby('test preparation course')[['math score','reading score','writing score']].max()

math score  reading score  writing score
test preparation course                                          
completed                       100            100            100
none                            100            100            100

5. The management needs your help to give bonus points to the top 25% of students based on their math score, so how will you help the management to achieve this.

In [22]:
top_students=Student_Performance.sort_values(by='math score',ascending=False)

In [23]:
top_students

gender race/ethnicity parental level of education         lunch  \
999    male        group E           bachelor's degree      standard   
996    male        group A                some college      standard   
515  female        group E                some college      standard   
517  female        group E          associate's degree      standard   
516  female        group E           bachelor's degree      standard   
..      ...            ...                         ...           ...   
2    female        group B            some high school  free/reduced   
51   female        group C            some high school  free/reduced   
3    female        group B                some college      standard   
1    female        group B                 high school  free/reduced   
0    female        group C            some high school  free/reduced   

    test preparation course  math score  reading score  writing score  
999               completed         100            100            100  
996               completed         100             96             86  
515                    none         100             92             97  
517                    none         100            100            100  
516                    none         100            100            100  
..                      ...         ...            ...            ...  
2                      none          18             32             28  
51                     none          13             41             51  
3                      none          11             38             32  
1                      none           8             24             23  
0                      none           0             17             10  

[1000 rows x 8 columns]

In [31]:
top_math_students=top_students.head(250)
top_math_students

gender race/ethnicity parental level of education         lunch  \
999    male        group E           bachelor's degree      standard   
996    male        group A                some college      standard   
515  female        group E                some college      standard   
517  female        group E          associate's degree      standard   
516  female        group E           bachelor's degree      standard   
..      ...            ...                         ...           ...   
856    male        group E            some high school      standard   
855    male        group E          associate's degree  free/reduced   
854    male        group D            some high school      standard   
853    male        group D          associate's degree  free/reduced   
852    male        group D                some college  free/reduced   

    test preparation course  math score  reading score  writing score  
999               completed         100            100            100  
996               completed         100             96             86  
515                    none         100             92             97  
517                    none         100            100            100  
516                    none         100            100            100  
..                      ...         ...            ...            ...  
856               completed          77             76             77  
855               completed          77             69             68  
854               completed          77             68             69  
853                    none          77             78             73  
852                    none          77             62             64  

[250 rows x 8 columns]

top_math_students are the 250 students that have the highest math score

In [34]:
math_Students=top_math_students.loc[:,'math score']

In [35]:
math_Students

999    100
996    100
515    100
517    100
516    100
      ... 
856     77
855     77
854     77
853     77
852     77
Name: math score, Length: 250, dtype: int64